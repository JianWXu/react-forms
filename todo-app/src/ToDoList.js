import { React, useState } from "react";
import { v4 as uuid } from "uuid";
import NewToDoForm from "./NewToDoForm";
import ToDo from "./ToDo";

function ToDoList() {
  const [toDoList, setToDoList] = useState([]);
  const addToDo = newToDo => {
    setToDoList(toDoList => [...toDoList, { ...newToDo, id: uuid() }]);
  };
  const deleteTask = index => {
    setToDoList(allTasks => allTasks.filter((task, i) => i !== index));
  };

  return (
    <div className="ToDoList">
      <h3>To Do List</h3>
      <NewToDoForm addToDo={addToDo} />
      {toDoList.map(({ task, id }, index) => (
        <div key={index}>
          <ToDo key={id} index={index} deleteTask={deleteTask} task={task} />
        </div>
      ))}
    </div>
  );
}

export default ToDoList;
