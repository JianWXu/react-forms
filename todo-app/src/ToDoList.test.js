import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  getByLabelText,
} from "@testing-library/react";
import ToDoList from "./ToDoList";

test("renders without crashing", () => {
  render(<ToDoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<ToDoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("Should add new task", async function () {
  const {
    getByPlaceholderText,
    queryByText,
    queryByLabelText,
    queryByDisplayValue,
  } = render(<ToDoList />);

  // checking to see if there are any tasks on the page right now
  const button = queryByText("Task Done!");
  expect(button).not.toBeInTheDocument();

  // queries the task InputDeviceInfo, clicks submitButton
  const taskInput = getByPlaceholderText("New Task");
  const submitButton = queryByText("Add Task to List");

  fireEvent.change(taskInput, { target: { value: "Cut Luna's Nails" } });
  fireEvent.click(submitButton);

  //   checks to see if a task is added to the page
  await waitFor(() => {
    const button = queryByText("Task Done!");
    expect(button).toBeInTheDocument();
  });

  // checks to see if delete task button works
  await waitFor(() => {
    const button = queryByText("Task Done!");
    fireEvent.click(button);
    expect(queryByText("Cut Luna's Nails")).not.toBeInTheDocument();
  });
});
