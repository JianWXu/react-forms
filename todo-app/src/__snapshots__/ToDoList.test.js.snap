// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoList"
  >
    <h3>
      To Do List
    </h3>
    <form>
      <label
        for="task"
      >
        Enter New Task: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="New Task"
        type="text"
        value=""
      />
      <button>
        Add Task to List
      </button>
    </form>
  </div>
</DocumentFragment>
`;
